pg_analytics/src/hooks/executor.rs

```diff
@@ -33,51 +28,52 @@ pub fn executor_run(
             .plannedstmt
             .current_query_string(CStr::from_ptr(query_desc.sourceText))?;
 
+        let (row_tables, col_tables) = rtable.table_lists()?;
+
         // Only use this hook for deltalake tables
         // Allow INSERTs to go through
         if rtable.is_null()
             || query_desc.operation == pg_sys::CmdType_CMD_INSERT
-            || !rtable.is_column()?
+            || col_tables.is_empty()
             // Tech Debt: Find a less hacky way to let COPY go through
             || query.to_lowercase().starts_with("copy")
         {
             prev_hook(query_desc, direction, count, execute_once);
             return Ok(());
         }
 
-        // Parse the query into a LogicalPlan
-        let logical_plan = match create_logical_plan(&query) {
-            Ok(logical_plan) => logical_plan,
-            // If DataFusion can't parse the query, let Postgres handle it
-            Err(_) => {
-                prev_hook(query_desc, direction, count, execute_once);
-                return Ok(());
+        // If tables of different types are both present in the query, federate the query.
+        if !row_tables.is_empty() && !col_tables.is_empty() {
+            let batches =
+                async_std::task::block_on(execute_federated_query(query, row_tables, col_tables))?;
+
+            match query_desc.operation {
+                pg_sys::CmdType_CMD_SELECT => select_with_batches(query_desc, batches),
+                _ => Err(ParadeError::NotSupported(NotSupported::Join(
```

Comment:
I think we should let non-SELECT queries go through. For instance, `INSERT INTO column_table SELECT * FROM row_table` is a supported command.

Reply:
So DML is not supported by federation yet and INSERT actually works in its current state via `prev_hook` - maybe because of table AM?

Reply:
yep, the table access method takes care of this